dart/lib/vm_service_lib.dart

```diff
@@ -132,7 +132,664 @@ Map<String, Function> _typeFactories = {
+  /// subscribed) error code is returned.
+  ///
+  /// See [Success].
+  Future<Success> streamCancel(String streamId);
+
+  /// The `streamListen` RPC subscribes to a stream in the VM. Once subscribed,
+  /// the client will begin receiving events from the stream.
+  ///
+  /// If the client is already subscribed to the stream, the `103` (Stream
+  /// already subscribed) error code is returned.
+  ///
+  /// The `streamId` parameter may have the following published values:
+  ///
+  /// streamId | event types provided
+  /// -------- | -----------
+  /// VM | VMUpdate
+  /// Isolate | IsolateStart, IsolateRunnable, IsolateExit, IsolateUpdate,
+  /// IsolateReload, ServiceExtensionAdded
+  /// Debug | PauseStart, PauseExit, PauseBreakpoint, PauseInterrupted,
+  /// PauseException, PausePostRequest, Resume, BreakpointAdded,
+  /// BreakpointResolved, BreakpointRemoved, Inspect, None
+  /// GC | GC
+  /// Extension | Extension
+  /// Timeline | TimelineEvents
+  ///
+  /// Additionally, some embedders provide the `Stdout` and `Stderr` streams.
+  /// These streams allow the client to subscribe to writes to stdout and
+  /// stderr.
+  ///
+  /// streamId | event types provided
+  /// -------- | -----------
+  /// Stdout | WriteEvent
+  /// Stderr | WriteEvent
+  ///
+  /// It is considered a `backwards compatible` change to add a new type of
+  /// event to an existing stream. Clients should be written to handle this
+  /// gracefully, perhaps by warning and ignoring.
+  ///
+  /// See [Success].
+  Future<Success> streamListen(String streamId);
+
+  /// Trigger a full GC, collecting all unreachable or weakly reachable objects.
+  @undocumented
+  Future<Success> collectAllGarbage(String isolateId);
+
+  /// `roots` is one of User or VM. The results are returned as a stream of
+  /// [_Graph] events.
+  @undocumented
+  Future<Success> requestHeapSnapshot(
+      String isolateId, String roots, bool collectGarbage);
+
+  /// Valid values for `gc` are 'full'.
+  @undocumented
+  Future<AllocationProfile> getAllocationProfile(String isolateId,
+      {String gc, bool reset});
+
+  /// Returns a ServiceObject (a specialization of an ObjRef).
+  @undocumented
+  Future<ObjRef> getInstances(String isolateId, String classId, int limit);
+  @undocumented
+  Future<Success> clearCpuProfile(String isolateId);
+
+  /// `tags` is one of UserVM, UserOnly, VMUser, VMOnly, or None.
+  @undocumented
+  Future<CpuProfile> getCpuProfile(String isolateId, String tags);
+  @undocumented
+  Future<Success> clearVMTimeline();
+  @undocumented
+  Future<Success> setVMTimelineFlags(List<String> recordedStreams);
+  @undocumented
+  Future<Response> getVMTimeline();
+  @undocumented
+  Future<Success> registerService(String service, String alias);
+}
+
+class VmServer {
+  final Stream<Map<String, Object>> requestStream;
+  final StreamSink<Map<String, Object>> responseSink;
+  final VmServiceInterface serviceImpl;
```

Comment:
Perhaps `serviceImplementation`? Flutter style tries to avoid abbreviations, which I've gotten used to in the devtools codebase.

Reply:
Sure, done (technically effective dart also says to avoid abbreviations).